### PR TITLE
feat: added toggle for dynamic booking in general settings #9060

### LINKED ISSUE #9060: [CAL-1743] Bring back the Dynamic Group booking opt-out toggle
The toggle that allows users to opt-out if the Dynamic group booking links is somehow lost and no longer resides in settings. We need to bring that back, as we know that some users might wish to opt-out of this feature.

We can place it within `/settings/my-account/general`

<sub>From [SyncLinear.com](https://synclinear.com) | [CAL-1743](https://linear.app/calcom/issue/CAL-1743/bring-back-the-dynamic-group-booking-opt-out-toggle)</sub>

### PR DESCRIPTION
**Pull request by sid0jack**
Fixes #9060 
The toggle that allows users to opt-out of the Dynamic group booking links wasn't available anymore and no longer resided in settings.
This pr adds back the toggle

- New feature (non-breaking change which adds functionality)
- Placed it within /settings/my-account/general

image

### CODE REVIEW

File: packages/trpc/server/routers/loggedInViewer/me.handler.ts [hunk 1]
```diff
@@ -37,5 +37,6 @@ export const meHandler = async ({ ctx }: MeOptions) => {
     theme: user.theme,
     hideBranding: user.hideBranding,
     metadata: user.metadata,
+    allowDynamicBooking: user.allowDynamicBooking,
```
Comment (alishaz-polymath) [inline packages/trpc/server/routers/loggedInViewer/me.handler.ts:40]: We don't really need to pull it here as well. Pulling it in the sessionMiddleware alone should suffice.

**Review by alishaz-polymath — DISMISSED** [review]
Just one change requested. The rest looks pretty good. 👍

File: packages/trpc/server/routers/loggedInViewer/me.handler.ts (hunk 1 above)
Comment (sid0jack) [reply packages/trpc/server/routers/loggedInViewer/me.handler.ts]: completed the change

**Comment by PeerRich** [comment]
CleanShot 2023-05-24 at 14 53 19@2x

this doesn't use our component.

can you use this instead:

CleanShot 2023-05-24 at 14 54 32@2x

**Comment by alishaz-polymath** [comment]
Also, the type-check is failing. It could be due to the missing changes in the 'me.handler.ts' but I wonder if there's a better fix. If not, feel free to re-add it 🙏 
We can follow up later and clean it up.

**Comment by sid0jack** [comment]
I will re-add it and check for a better solution, which I will try to take up in the next pr.

**Comment by alishaz-polymath** [comment]
In the mean time, please address the change requested by @PeerRich 🙏 
Once again, thank you for the contribution :)

File: apps/web/pages/settings/my-account/general.tsx [hunk 2]
```diff
@@ -213,6 +215,21 @@ const GeneralView = ({ localeProp, user }: GeneralViewProps) => {
           </>
         )}
       />
+      <div className="mt-8">
+        <Controller
+          name="allowDynamicBooking"
+          control={formMethods.control}
+          render={() => (
+            <SettingsToggle
+              title={t("allow_dynamic_booking")}
```
Comment (alishaz-polymath) [inline apps/web/pages/settings/my-account/general.tsx:224]:
```suggestion
              title={t("dynamic_booking")}
              description{t("allow_dynamic_booking")}
```
Yep, the styling is off. You can do something like this. Please note that `dynamic_booking` doesn't exist in the locale yet, so you'd need to add it to the 'en' locale as well. I believe that should do it. Unless @PeerRich has something to add to this.

**Review by alishaz-polymath — CHANGES_REQUESTED** [review]

File: apps/web/pages/settings/my-account/general.tsx (hunk 2 above)
Comment (alishaz-polymath) [reply apps/web/pages/settings/my-account/general.tsx:224]: It would look something like this:
image

Comment (alishaz-polymath) [reply apps/web/pages/settings/my-account/general.tsx:224]: The controls for the Logged-In theme is currently being reviewed, if I'm not mistaken. Currently, it just picks system default theme for the logged-in section. The users will be able to control it better after a PR that is under review, if I'm not mistaken. I mean, the light mode should be fairly similar here. I believe we need the title and description in the toggle and that just adjusts the styling well enough!

**Comment by PeerRich** [comment]
can fix once feedback addressed

**Comment by alishaz-polymath** [comment]
hi @sid0jack Is there any update on this?

**Comment by sid0jack** [comment]
Hi, I was busy. I will try to address the feedback and close this asap

**Comment by sid0jack** [comment]
image

**Review by alishaz-polymath — CHANGES_REQUESTED** [review]
Please remove yarn lockfile

**Comment by sid0jack** [comment]
@alishaz-polymath I have removed the lockfile but for some reason the lint checker is failing, it is working fine locally. Most probably it is getting timed out.

**Review by alishaz-polymath — APPROVED** [review]
Looks good!
Thank you for your contribution 🙏 
This should be merged when the Checks pass. The Lint check currently failing is not related to this PR, so don't worry about it. 🚀

**Pull request merged**